src/aleph/vm/pool.py

```diff
@@ -170,10 +187,14 @@ def forget_vm(self, vm_hash: ItemHash) -> None:
             pass
 
     async def stop(self):
-        """Stop all VMs in the pool."""
-
+        """Stop ephemeral VMs in the pool."""
         # Stop executions in parallel:
-        await asyncio.gather(*(execution.stop() for vm_hash, execution in self.executions.items()))
+        await asyncio.gather(*(execution.stop() for vm_hash, execution in self.get_ephemeral_executions()))
```

Comment:
What would be the procedure to stop all running VMs, including persistent VMs ?

Reply:
This method is just called to stop the running VMs when the orchestrator goes down, and this should not stop the persistent VMs.
We don't have a method to stop all VMs, and also I think that we don't need it.

Reply:
Should we at least add a CLI or script that stops persistent VMs in case we need it ?

Reply:
Move to https://github.com/aleph-im/aleph-vm/issues/505